### PR TITLE
Darkel/Wanted update.

### PR DESCRIPTION
**Pull request by gennariarmando**

### CODE REVIEW

File: src/control/Darkel.cpp [hunk 1]
```diff
@@ -1,12 +1,320 @@
...
+	}
+	else
+		return 0;
+}
+
+
+#if 0
 WRAPPER void CDarkel::DrawMessages(void) { EAXJMP(0x420920); }
+#else
+void CDarkel::DrawMessages()
```
Comment (aap) [inline src/control/Darkel.cpp]: This looks quite different from the original function

File: src/control/Darkel.cpp [hunk 2]
```diff
@@ -1,12 +1,320 @@
...
+int8 &CDarkel::bProperKillFrenzy = *(int8*)0x95CD98;
+eKillFrenzyStatus &CDarkel::Status = *(eKillFrenzyStatus*)0x95CCB4;
+int16 *CDarkel::RegisteredKills = (int16*)0x6EDBE0;
+int32 &CDarkel::ModelToKill = *(int32*)0x8F2C78;
+int32 &CDarkel::ModelToKill2 = *(int32*)0x885B40;
+int32 &CDarkel::ModelToKill3 = *(int32*)0x885B3C;
+int32 &CDarkel::ModelToKill4 = *(int32*)0x885B34;
+wchar *CDarkel::pStartMessage = (wchar*)0x8F2C08;
+
+int32 CDarkel::CalcFade(uint32 time, int32 min, uint32 max) {
```
Comment (aap) [inline src/control/Darkel.cpp]: this returns a uint8. Also maybe rename min, max to start, end

File: src/control/Darkel.cpp [hunk 3]
```diff
@@ -1,12 +1,320 @@
...
+	if (time >= min && time <= max) {
+		if (time >= min + 500) {
+			if (time <= max - 500)
+				return -1;
```
Comment (aap) [inline src/control/Darkel.cpp]: 255

File: src/control/Darkel.cpp [hunk 4]
```diff
@@ -1,12 +1,320 @@
...
+
+
+eKillFrenzyStatus CDarkel::ReadStatus()
+{
+	return Status;
+}
+
+int32 CDarkel::RegisterCarBlownUpByPlayer(eKillFrenzyStatus status)
+{
+	return 0;
```
Comment (aap) [inline src/control/Darkel.cpp]: don't put this function in the code if it's not implemented, put it in a comment or #if 0

File: src/control/Darkel.cpp [hunk 5]
```diff
@@ -1,12 +1,320 @@
...
+}
+
+void CDarkel::RegisterKillByPlayer(int32 modelid, eWeaponType weapontype, bool flag)
+{
+
```
Comment (aap) [inline src/control/Darkel.cpp]: same as above

File: src/control/Darkel.cpp [hunk 6]
```diff
@@ -1,12 +1,320 @@
...
+void CDarkel::RegisterKillNotByPlayer()
+{
+	++CStats::NumberKillFrenziesPassed;
+}
+
+void CDarkel::ResetModelsKilledByPlayer()
+{
+	for (int i = 0; i < 200; i++) {
+		for (int j = 0; j < 8; j++)
```
Comment (aap) [inline src/control/Darkel.cpp]: remove j loop

File: src/control/Darkel.cpp [hunk 7]
```diff
@@ -1,12 +1,320 @@
...
+			RegisteredKills[i + j] = 0;
+	};
+}
+
+void CDarkel::ResetOnPlayerDeath()
+{
+	eWeaponType &EWeaponType = CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_weapons[CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_currentWeapon].m_eWeaponType;
```
Comment (aap) [inline src/control/Darkel.cpp]: this code is inside the if in the game

File: src/control/Darkel.cpp [hunk 8]
```diff
@@ -1,12 +1,320 @@
...
+	int32 &CurrentWeapon = CWeaponInfo::GetWeaponInfo(FindPlayerPed()->GetWeapon()->m_eWeaponType)->m_nModelId;
+	int32 &CurrentDarkelWeapon = CDarkel::WeaponType;
+	uint32 &TotalAmmo = CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_weapons[CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_currentWeapon].m_nAmmoTotal;
+
+	if (Status == KILLFRENZY_ONGOING) {
```
Comment (aap) [inline src/control/Darkel.cpp]: better: `if (Status != KILLFRENZY_ONGOING) return;`

File: src/control/Darkel.cpp [hunk 9]
```diff
@@ -1,12 +1,320 @@
...
+		}
+	}
+}
+
+#if 0
+WRAPPER void CDarkel::StartFrenzy(eWeaponType weaponType, int32 time, int16 kill, int32 modelId0, wchar *text, int32 modelId2, int32 modelId3, int32 modelId4, bool standardSound, bool needHeadShot) { EAXJMP(0x4210E0); }
+#else
+void CDarkel::StartFrenzy(eWeaponType weaponType, int32 time, int16 kill, int32 modelId0, wchar *text, int32 modelId2, int32 modelId3, int32 modelId4, bool standardSound, bool needHeadShot)
+{
+	eWeaponType &EWeaponType = CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_weapons[CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_currentWeapon].m_eWeaponType;
```
Comment (aap) [inline src/control/Darkel.cpp]: this code isn't at the top here in the game

File: src/control/Darkel.cpp [hunk 10]
```diff
@@ -1,12 +1,320 @@
...
+		}
+	}
+	if (CDarkel::bStandardSoundAndMessages)
+		DMAudio.PlayFrontEndSound(SOUND_RAMPAGE_START, 0);
+}
+#endif
+
+void CDarkel::Update()
+{
+	eWeaponType &EWeaponType = CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_weapons[CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_currentWeapon].m_eWeaponType;
```
Comment (aap) [inline src/control/Darkel.cpp]: again, not here.

File: src/control/Darkel.cpp [hunk 11]
```diff
@@ -1,12 +1,320 @@
...
+	int32 &CurrentWeapon = CWeaponInfo::GetWeaponInfo(FindPlayerPed()->GetWeapon()->m_eWeaponType)->m_nModelId;
+	int32 &CurrentDarkelWeapon = CDarkel::WeaponType;
+	uint32 &TotalAmmo = CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_weapons[CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_currentWeapon].m_nAmmoTotal;
+
+	if (Status == KILLFRENZY_ONGOING) {
```
Comment (aap) [inline src/control/Darkel.cpp]: again, better return when !=

File: src/control/Darkel.h [hunk 12]
```diff
@@ -1,11 +1,48 @@
 #pragma once
+#include "Weapon.h"
+
+enum eKillFrenzyStatus
+{
+	KILLFRENZY_INIT,
```
Comment (aap) [inline src/control/Darkel.h]: INIT sounds weird...maybe NONE or INACTIVE or something?

File: src/control/Darkel.cpp (hunk 3 above)
Comment (gennariarmando) [reply src/control/Darkel.cpp]: What's 255 here?

File: src/control/Darkel.cpp (hunk 1 above)
Comment (gennariarmando) [reply src/control/Darkel.cpp]: Has been cleaned up from unused stuff.

File: src/control/Darkel.cpp (hunk 3 above)
Comment (aap) [reply src/control/Darkel.cpp]: that -1 is a 255

File: src/control/Darkel.cpp (hunk 1 above)
Comment (aap) [reply src/control/Darkel.cpp]: cleaning up isn't what we want to do at this stage. it's much more important to have code close to the original game.

**Pull request merged**